pkgs/applications/editors/emacs/elisp-packages/manual-packages/prolog/default.nix

```diff
@@ -1,22 +1,18 @@
-{ lib, stdenv, fetchurl }:
+{ lib, trivialBuild, fetchurl, emacs }:
 
-stdenv.mkDerivation {
+trivialBuild {
   pname = "prolog-mode";
   version = "1.28";
 
   src = fetchurl {
-    url = "http://bruda.ca/_media/emacs/prolog.el";
+    url = "https://bruda.ca/_media/emacs/prolog.el";
     sha256 = "ZzIDFQWPq1vI9z3btgsHgn0axN6uRQn9Tt8TnqGybOk=";
   };
 
-  buildCommand = ''
-    mkdir -p $out/share/emacs/site-lisp/
-    cp $src $out/share/emacs/site-lisp/prolog.el
-  '';
-
   meta = {
     homepage = "https://bruda.ca/emacs/prolog_mode_for_emacs/";
     description = "Prolog mode for Emacs";
     license = lib.licenses.gpl2Plus;
+    inherit (emacs.meta) platforms;
```

Comment:
I think this is the default
https://github.com/NixOS/nixpkgs/blob/03cb41b569da52d2f639e99e1f35cfb89f0ef47e/pkgs/build-support/emacs/generic.nix#L22

Reply:
Oh, you are right. https://github.com/NixOS/nixpkgs/blob/a991941a3abdacd75d6d70cd1e2492d28a775d6f/pkgs/build-support/emacs/generic.nix#L22 
I have copied it from somewhere else. Then I will remove it from other places as well in the future.